subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -515,18 +519,21 @@ public boolean isPresent() {
         }
 
         @Override
-        public void configure(final Action<? super Task> action) {
-            configureLater(name, action);
+        public void configure(final Action<? super T> action) {
+            configureLater(name, new Action<Task>() {
```

Comment:
I think we could also just do this (delete `configureLater`):
```
configureEachLater(new Action<Task>() {
            @Override
            public void execute(Task task) {
                if (task.getName().equals(name)) {
                    action.execute((T)task);
                }
            }
        });
```

Reply:
Good point. I saw `configureLater` was public and wrongly assumed it was on the public interface.